hoomd/logging.py

```diff
@@ -1,74 +1,252 @@
+    from strings to the enum wherever necessary in the API. This class is
+    documented to show users what types of quantities can be logged, and what
+    flags to use for limiting what data is logged, user specified logged
+    quantites, and custom actions (`hoomd.custom.Action`).
+
+    Flags:
+        scalar: `float` or `int` objects (i.e. numbers)
+
+        sequence: sequence (e.g. `list`, `tuple`, `numpy.ndarray`) of numbers of
+        the same type.
+
+        string: a single Python `str` object
+
+        strings: a sequence of Python `str` objects
+
+        object: any Python object outside a sequence, string, or scalar.
+
+        angle: per-angle quantity
+
+        bond: per-bond quantity
+
+        constraint: per-constraint quantity
+
+        dihedral: per-dihedral quantity
+
+        improper: per-improper quantity
+
+        pair: per-pair quantity
+
+        particle: per-particle quantity
+
+        state: internal flag for specifying object's internal state
+
+        ALL: a combination of all other flags
+
+        NONE: represents no flag
+    """
+    NONE = 0
+    scalar = auto()
+    sequence = auto()
+    string = auto()
+    strings = auto()
+    object = auto()
+    angle = auto()
+    bond = auto()
+    constraint = auto()
+    dihedral = auto()
+    improper = auto()
+    pair = auto()
+    particle = auto()
+    state = auto()
 
     @classmethod
-    def log(cls, func=None, is_property=True, flag='scalar'):
-        def helper(func):
-            name = func.__name__
-            if name in cls._meta_export_dict:
-                raise KeyError(
-                    "Multiple loggable quantities named {}.".format(name))
-            cls._meta_export_dict[name] = flag
-            if is_property:
-                return property(func)
+    def any(cls, flags):
+        def from_str(flag):
+            if isinstance(flag, str):
+                return cls[flag]
             else:
-                return func
-        if func is None:
-            return helper
-        else:
-            return helper(func)
-
-    def __new__(cls, name, base, dct):
-        new_cls = super().__new__(cls, name, base, dct)
-        log_dict = dict()
-        for name, flag in cls._meta_export_dict.items():
-            log_dict[name] = LoggerQuantity(name, new_cls, flag)
-        if hasattr(new_cls, '_export_dict'):
-            old_dict = deepcopy(new_cls._export_dict)
-            for key, value in old_dict.items():
-                old_dict[key] = value.update_cls(new_cls)
-            old_dict.update(log_dict)
-            new_cls._export_dict = old_dict
+                return flag
+
+        return reduce(
+            lambda x, y: from_str(x) | from_str(y), flags, TypeFlags.NONE)
+
+    @classmethod
+    def _get_string_list(cls, flag):
+        return [mem.name for mem in cls.__members__.values() if mem in flag]
+
+
+TypeFlags.ALL = TypeFlags.any(TypeFlags.__members__.values())
```

Comment:
Largely to just allow access without needing `()`. A property won't work since you would only be able to use `ALL` on an instance of the class then.

Reply:
What if you instead defined `any` with a default `flags=None` that did this?

Reply:
I think that is fine, but I don't see a reason not to have this be a member of `TypeFlags` since any combination of `TypeFlags` will still be a `TypeFlags` object.